### PR TITLE
Retry Only If the Status Code Is Retryable

### PR DESCRIPTION
**Pull request by VikramVuppla**
Signed-off-by: Vikram Vuppla naga.vicky@gmail.com

Fixes #

Why:
- Currently the retry is happening regardless of the response status code is retryable or not, remove the buggy retry logic

## Proposed Changes

- :bug: Fix bug

### CODE REVIEW

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java [hunk 1]
```diff
@@ -135,10 +135,6 @@ private Future<HttpResponse<Buffer>> send(final CloudEvent event, final int retr
           }
         }
 
-        if (retryCounter < consumerVerticleContext.getEgressConfig().getRetry()) {
-          return retry(retryCounter, event);
-        }
```
Comment (pierDipi) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java:140]: This is handling the case where we have a failure but without a response, I think we're missing a return at line 135 when this `if (isRetryableStatusCode(response.statusCode()) && retryCounter < consumerVerticleContext.getEgressConfig().getRetry())` is false

**Review by pierDipi — COMMENTED** [review]
Thanks for pointing this out @VikramVuppla!

**Comment by VikramVuppla** [comment]
thank you @pierDipi for the review, addressed the review comments, please re-review

**Review by pierDipi — COMMENTED** [review]
Thanks!

Can we add a new unit test covering this case in WebClientCloudEventSenderTest (see example test `shouldRetry`?

Also, EasyCLA complains for the unsigned Linux Foundation CLA: https://github.com/knative-sandbox/eventing-kafka-broker/pull/2793#issuecomment-1319316385

**Comment by VikramVuppla** [comment]
@pierDipi EasyCLA is now covered and added the unit test case, could you please re-review

**Review by pierDipi — APPROVED** [review]
Thanks @VikramVuppla !

/lgtm
/approve
/retest-required

**Pull request merged**